### PR TITLE
Add support for Rails 7.2 !!!!!!

### PR DESCRIPTION
**Pull request by reeganviljoen**
### What are you trying to accomplish?

- [x] Add support for Rails 7.2 and remove explicit [YJITT testing as its the new default](https://edgeguides.rubyonrails.org/7_2_release_notes.html#enable-yjit-by-default-if-running-ruby-3-3)

https://rubyonrails.org/2024/8/10/Rails-7-2-0-has-been-released

### CODE REVIEW

**Comment by reeganviljoen** [comment]
@joelhawksley it seems main, head is failing again, I will investigate tomorrow

**Comment by reeganviljoen** [comment]
@joelhawksley the failing head and main tests have nothing to do with this would you mind taking a look

**Review by boardfish — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,43 +35,40 @@ jobs:
           - ruby_version: "3.0"
             rails_version: "6.1"
             mode: "capture_patch_enabled"
-            yjit_mode: '0'
+
```
Comment (joelhawksley) [inline .github/workflows/ci.yml:38]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -35,43 +35,40 @@ jobs:
...
           - ruby_version: "3.0"
             rails_version: "6.1"
             mode: "capture_patch_disabled"
-            yjit_mode: '0'
+
```
Comment (joelhawksley) [inline .github/workflows/ci.yml:42]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -35,43 +35,40 @@ jobs:
...
           - ruby_version: "3.1"
             rails_version: "7.0"
             mode: "capture_patch_enabled"
-            yjit_mode: '1'
           - ruby_version: "3.1"
             rails_version: "7.0"
             mode: "capture_patch_disabled"
-            yjit_mode: '0'
+
```
Comment (joelhawksley) [inline .github/workflows/ci.yml:49]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -35,43 +35,40 @@ jobs:
...
           - ruby_version: "3.2"
             rails_version: "7.1"
             mode: "capture_patch_enabled"
-            yjit_mode: '0'
+
```
Comment (joelhawksley) [inline .github/workflows/ci.yml:53]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -35,43 +35,40 @@ jobs:
...
             rails_version: "7.1"
             mode: "capture_patch_enabled"
-            yjit_mode: '1'
           - ruby_version: "3.3"
             rails_version: "7.1"
             mode: "capture_patch_disabled"
-            yjit_mode: '1'
+          - ruby_version: "3.3"
+            rails_version: "7.2"
+            mode: "capture_patch_disabled"
```
Comment (joelhawksley) [inline .github/workflows/ci.yml:65]:
```suggestion
          - ruby_version: "3.3"
            rails_version: "7.2"
            mode: "capture_patch_disabled"
         - ruby_version: "3.3"
            rails_version: "7.2"
            mode: "capture_patch_enabled"
```

File: docs/CHANGELOG.md [hunk 6]
```diff
@@ -30,6 +30,10 @@ nav_order: 5
 
     *Reegan Viljoen*
 
+* Add support for Rails 7.2
```
Comment (joelhawksley) [inline docs/CHANGELOG.md:33]:
```suggestion
* Add support for Rails 7.2.
```

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -35,43 +35,39 @@ jobs:
...
+         - ruby_version: "3.3"
```
Comment (joelhawksley) [inline .github/workflows/ci.yml:62]:
```suggestion
          - ruby_version: "3.3"
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -35,43 +35,39 @@ jobs:
...
             rails_version: "7.1"
             mode: "capture_patch_disabled"
-            yjit_mode: '1'
           - ruby_version: "3.3"
             rails_version: "7.1"
             mode: "capture_patch_enabled"
-            yjit_mode: '1'
           - ruby_version: "3.3"
             rails_version: "7.1"
             mode: "capture_patch_disabled"
```
Comment (joelhawksley) [inline .github/workflows/ci.yml:58]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -35,43 +35,39 @@ jobs:
...
             rails_version: "7.1"
             mode: "capture_patch_enabled"
-            yjit_mode: '0'
           - ruby_version: "3.2"
             rails_version: "7.1"
             mode: "capture_patch_disabled"
-            yjit_mode: '1'
           - ruby_version: "3.3"
             rails_version: "7.1"
             mode: "capture_patch_enabled"
```
Comment (joelhawksley) [inline .github/workflows/ci.yml:55]:
```suggestion
```

**Pull request merged**